flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkITCase.java

```diff
@@ -0,0 +1,195 @@
+@Testcontainers
+@ExtendWith(MiniClusterExtension.class)
+public class DynamoDbDynamicSinkITCase {
+    private static final String PARTITION_KEY = "partition_key";
+    private static final String SORT_KEY = "sort_key";
+    private static DynamoDBHelpers dynamoDBHelpers;
+    private static String testTableName;
+    private static StreamExecutionEnvironment env;
+
+    // shared between test methods
+    @Container
+    public static final DynamoDbContainer LOCALSTACK =
+            new DynamoDbContainer(DockerImageName.parse(DockerImageVersions.DYNAMODB))
+                    .withCommand("-jar DynamoDBLocal.jar -inMemory -sharedDb")
+                    .withNetwork(Network.newNetwork())
+                    .withNetworkAliases("dynamodb");
+
+    @BeforeEach
+    public void setup() throws URISyntaxException {
+        testTableName = UUID.randomUUID().toString();
+        env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.setParallelism(1);
+
+        dynamoDBHelpers = new DynamoDBHelpers(LOCALSTACK.getHostClient());
+    }
+
+    @Test
+    public void testSQLSink() throws ExecutionException, InterruptedException {
+
+        int expectedNumOfElements = 50;
+
+        dynamoDBHelpers.createTable(testTableName, PARTITION_KEY, SORT_KEY);
+        StreamTableEnvironment streamTableEnvironment =
+                StreamTableEnvironment.create(env, EnvironmentSettings.newInstance().build());
+
+        final String createTableStmt = getCreateTableStmt();
+        streamTableEnvironment.executeSql(createTableStmt);
+
+        final String datagenStmt = getDatagenStmt(expectedNumOfElements);
+        streamTableEnvironment.executeSql(datagenStmt);
+
+        final String insertSql = "INSERT INTO dynamo_db_table SELECT * from datagen;";
+        streamTableEnvironment.executeSql(insertSql).await();
+
+        Assertions.assertThat(dynamoDBHelpers.getItemsCount(testTableName))
+                .isEqualTo(expectedNumOfElements);
+    }
+
+    @Test
+    public void testTableAPISink() throws ExecutionException, InterruptedException {
+
+        int expectedNumOfElements = 50;
+
+        dynamoDBHelpers.createTable(testTableName, PARTITION_KEY, SORT_KEY);
+        StreamTableEnvironment streamTableEnvironment =
+                StreamTableEnvironment.create(env, EnvironmentSettings.newInstance().build());
+
+        final String createTableStmt = getCreateTableStmt();
+        streamTableEnvironment.executeSql(createTableStmt);
+
+        final String datagenStmt = getDatagenStmt(expectedNumOfElements);
+        streamTableEnvironment.executeSql(datagenStmt);
+
+        Table resultTable = streamTableEnvironment.sqlQuery("SELECT * FROM datagen");
+
+        resultTable.executeInsert("dynamo_db_table").await();
+
+        Assertions.assertThat(dynamoDBHelpers.getItemsCount(testTableName))
+                .isEqualTo(expectedNumOfElements);
```

Comment:
Refering to Firehose test, we used to see intermittent failures due to delays between sending records downstream and observing them from LocalStack, is this not needed for DDB?

Reply:
The line before is a synchronous operation "resultTable.executeInsert("dynamo_db_table").await();" which should wait until all records are successfully sent. So the delay shouldn't cause us failures here unless the await timeouts